wgpu-hal/src/metal/mod.rs

```diff
@@ -388,8 +390,23 @@ impl crate::Queue<Api> for Queue {
     }
 
     unsafe fn get_timestamp_period(&self) -> f32 {
-        // TODO: This is hard, see https://github.com/gpuweb/gpuweb/issues/1325
-        1.0
+        if !self.shared.private_caps.supports_timestamp_period {
+            return 1.0;
+        }
+
+        use objc::{msg_send, runtime::Object, sel, sel_impl};
+        let timestamps0 = [std::ptr::null_mut::<Object>(); 2];
```

Comment:
why are we creating pointers to objects here? these are meant to be numbers

Reply:
wgpu CI check suggested change it to this：
```sh
error: `0 as *mut _` detected
[27](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:27)
   --> wgpu-hal/src/metal/mod.rs:395:31
[28](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:28)
    |
[29](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:29)
395 |             let timestamps = [0_u64 as *mut Object; 4];
[30](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:30)
    |                               ^^^^^^^^^^^^^^^^^^^^ help: try: `std::ptr::null_mut::<Object>()`
[31](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:31)
    |
[32](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:32)
    = note: `-D clippy::zero-ptr` implied by `-D warnings`
[33](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:33)
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#zero_ptr
[34](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:34)

[35](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:35)
error: aborting due to previous error
[36](https://github.com/gfx-rs/wgpu/runs/5465721991?check_suite_focus=true#step:12:36)
```

Reply:
I think I know what your mean now, will revise it